File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    d = distance({x, y})
    cond do
      d > 10 -> 0
      d > 5 -> 1
      d > 1 -> 5
      true -> 10
    end
  end

  defp distance({x, y}), do: :math.sqrt(x * x + y * y)
end

File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y = case distFromCircle x y of
    d | d <= 1    -> 10
      | d <= 5    -> 5
      | d <= 10   -> 1
      | otherwise -> 0

distFromCircle :: Float -> Float -> Float
distFromCircle x y = sqrt (x * x + y * y)